src/formatDate.js

```diff
@@ -7,8 +7,56 @@
  *
  * @returns {string}
  */
+
 function formatDate(date, fromFormat, toFormat) {
-  // write code here
+  const dateElements = date.split(fromFormat[fromFormat.length - 1]);
+  const newFromDate = {};
+
+  for (let i = 0; i < fromFormat.length; i++) {
+    const format = fromFormat[i];
+
+    if (format === 'YYYY') {
+      newFromDate.year = +dateElements[i];
+    } else if (format === 'YY') {
+      newFromDate.year = +dateElements[i];
+
+      if (newFromDate.year < 30) {
+        newFromDate.year += 2000;
+      } else {
+        newFromDate.year += 1900;
+      }
+    } else if (format === 'MM') {
+      newFromDate.month = dateElements[i];
+    } else if (format === 'DD') {
+      newFromDate.day = dateElements[i];
+    }
+  }
```

Comment:
The loop can be optimized by using a single `if-else` chain or a `switch` statement to handle the assignment of year, month, and day to `newFromDate`. This will make the code more concise and easier to read.